ignite/internal/plugin/consumer.go

```diff
@@ -0,0 +1,39 @@
+package plugininternal
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/ignite/cli/v28/ignite/pkg/errors"
+	"github.com/ignite/cli/v28/ignite/services/plugin"
+)
+
+// TODO use released version of app-consumer
+const consumerPlugin = "github.com/ignite/apps/consumer"
+
+// ConsumerWriteGenesis writes validators in the consumer module genesis.
+// NOTE(tb): Using a plugin for this task avoids having the interchain-security
+// dependency in Ignite.
+func ConsumerWriteGenesis(ctx context.Context, c plugin.Chainer) error {
+	_, err := Execute(ctx, consumerPlugin, []string{"writeGenesis"}, plugin.WithChain(c))
+	if err != nil {
+		return errors.Errorf("execute consumer plugin 'writeGenesis': %w", err)
+	}
+	return nil
+}
+
+// ConsumerIsInitialized returns true if the consumer chain's genesis c has
+// a consumer module entry with an initial validator set.
+// NOTE(tb): Using a plugin for this task avoids having the interchain-security
+// dependency in Ignite.
+func ConsumerIsInitialized(ctx context.Context, c plugin.Chainer) (bool, error) {
+	out, err := Execute(ctx, consumerPlugin, []string{"isInitialized"}, plugin.WithChain(c))
```

Comment:
can we communicate with the plugin by GRPC?

Reply:
cc @tbruyelle

Reply:
I didn't participate in this enhancement, I don't know how to do it. What's the advantage over the current encoding?

Reply:
I didn't participate either, but we can communicate with ignite and the app through GRPC. @jeronimoalbi was more aware. Can we do it here? And will it benefit us more?

Reply:
Shall we not ship this instead and make optimization later on? That's a great feature for v28.3.0